full-service/src/db/txo.rs

```diff
@@ -561,36 +625,39 @@ impl TxoModel for Txo {
             txo: txo.clone(),
             received_to_account: None,
             received_to_assigned_subaddress: None,
-            spent_from_account: None,
+            secreted_from_account: None,
         };
 
         for account_txo_status in account_txo_statuses {
             match account_txo_status.txo_type.as_str() {
                 TXO_MINTED => {
-                    txo_details.spent_from_account = Some(account_txo_status.clone());
+                    txo_details.secreted_from_account = Some(account_txo_status.clone());
+                    // Note: Minted & Unspent means that this Txo was also
+                    // received, and is either change, or a
+                    // Txo that we went to ourselves.
                 }
                 TXO_RECEIVED => {
                     txo_details.received_to_account = Some(account_txo_status.clone());
-                    // Get the subaddress details if assigned
-                    let assigned_subaddress = if let Some(subaddress_index) = txo.subaddress_index {
-                        let account: Account =
-                            Account::get(&AccountID(account_txo_status.account_id_hex), conn)?;
-                        let account_key: AccountKey = mc_util_serial::decode(&account.account_key)?;
-                        let subaddress = account_key.subaddress(subaddress_index as u64);
-                        let subaddress_b58 = b58_encode(&subaddress)?;
-
-                        Some(AssignedSubaddress::get(&subaddress_b58, conn)?)
-                    } else {
-                        None
-                    };
-                    txo_details.received_to_assigned_subaddress = assigned_subaddress;
                 }
                 _ => {
                     return Err(WalletDbError::UnexpectedTransactionTxoType(
                         account_txo_status.txo_type,
                     ))
                 }
             }
+            // Get the subaddress details if assigned
```

Comment:
Moved here because a minted & unspent transaction can have an assigned subaddress (this is when sent to self, or as change)